### ISSUE TITLE
Additive inverse of number literals

### ISSUE THREAD

**Issue by m93a** [Microsoft/TypeScript#30914]
## Search Terms
Using unary minus on variable that is of number-literal type. Negative number literals.

## Suggestion
This is a specific case of #26382. My proposal is simply that if we apply unary minus on an expression whose type is a union of number literal types, the result should be a union of their additive inverses:
```typescript
declare var a: 1 | 2 | 3;
const b = -a; // should be -1 | -2 | -3
```

Furthermore we could add a unary minus type operator:
```typescript
-(42) === -42
-(A | B) === -A | -B
-number === number
-(not number) // error
```

Both should be very easy to implement.

## Use Cases
```typescript
function compare(a: C, b: C): -1 | 0 | 1;

function reverseCompare(a: C, b: C): -1 | 0 | 1
{
  return -compare(a, b); // shouldn't throw an error
}
```

The type operator would be just for sake of completeness

## Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

**Comment by keitzer** [comment]
Our example code:


```
export const SpacingUnit = {
	/** xxs - 4 unit spacing */
	xxs_4: 4,
	/** xs - 8 unit spacing */
	xs_8: 8,
	/** sm - 12 unit spacing */
	sm_12: 12,
	// ... others
} as const;

export type BoxSpacingUnit =
	/* others */
	| -12
	| -8
	| -4
	| 0
	| 4
	| 8
	| 12
	/* others */
	| `-${keyof typeof SpacingUnit}`
	| keyof typeof SpacingUnit;

interface BoxStyleProps {
	// Margin
	marginTop?: BoxSpacingUnit;
	marginBottom?: BoxSpacingUnit;
	marginLeft?: BoxSpacingUnit;
	marginRight?: BoxSpacingUnit;
	marginHorizontal?: BoxSpacingUnit;
	marginVertical?: BoxSpacingUnit;
	margin?: BoxSpacingUnit;

	// ... others...
}
```

![Image](https://github.com/user-attachments/assets/89f593c2-629f-4dba-b929-5854594220f3)

Adding support for this will be helpful for Design Systems that constrain possible styling values.
